components/automate-cli/cmd/chef-automate/config.go

```diff
@@ -64,6 +64,10 @@ const (
 var configValid = "Config file must be a valid %s config"
 
 func init() {
+	err := fileutils.CreateHabTmpDir()
```

Comment:
Instead of calling `fileutils.CreateHabTmpDir()` from init(), 
call from `doDeployWork` function of below files

- `automate/components/automate-cli/cmd/chef-automate/automateHADeployExistingInfra.go` for On-Prem 
- `automate/components/automate-cli/cmd/chef-automate/automateHAAwsDeployment.go` for AWS

Reply:
When I removed the creation of `/hab/tmp` dir from `config.go`'s `init()` function, and moved the creation of the latter into the 2 files you suggested, I faced the following errors while adding/removing nodes:
`UnknownError: error on removing output header in fetched config: open /hab/tmp/temp3844032987: no such file or directory`

The `hab` directory is not being created when performing those commands. The steps I went through:
1) Made the changes you suggested locally
2) Rebuilt the automate-cli code
3) Uploaded the artifact pkg to hab
4) Removed the chef-automate binary
5) Installed my chef-automate pkg from habitat on the bastion host and created a symlink to it.
6) Created a new empty EC2 instance
7) Executed the command: `chef-automate node add -A 10.0.14.173`

I performed the same steps but with my previous code where I created the `/hab/tmp` directory in an `init()` function inside `config.go` and I had no errors. I also deleted the `/hab/tmp` folder before each command, and it was recreated:
<img width="583" alt="image" src="https://github.com/chef/automate/assets/160588026/118bc3e0-f8be-42dc-a289-5bd6b9d5e293">
<img width="584" alt="image" src="https://github.com/chef/automate/assets/160588026/0af2d88c-956b-4d2a-8591-7bf7182f5823">